libs/cri-api-service/src/main/java/uk/gov/di/ipv/core/library/criapiservice/CriApiService.java

```diff
@@ -138,7 +138,7 @@ private BearerAccessToken fetchAccessToken(
                         errorObject.getDescription(),
                         errorObject.getHTTPStatusCode());
                 throw new CriApiException(
-                        HTTPResponse.SC_BAD_REQUEST, ErrorResponse.INVALID_TOKEN_REQUEST);
+                        httpResponse.getStatusCode(), ErrorResponse.INVALID_TOKEN_REQUEST);
```

Comment:
The other noticeable hardcoding is for `fetchVerifiableCredential`, which can be updated to not be hardcoded either (since the logic is duplicated in the catch statement in `ProcessCriCallbackHandler`). Other hardcoded values come from thrown exceptions, which makes sense.

I have walked through all paths from where I am suggesting to update: `fetchAccessToken` & `fetchVerifiableCredential`. There are no functional differences with updating these status codes, it's just that `core-front` gets a more accurate status code - the intention behind the refactor.

That being said, it has a low reward/risk ratio to do this refactor so I will remove it from this PR.